app/api.py

```diff
@@ -0,0 +1,195 @@
+import uvicorn
+from fastapi import FastAPI, File, UploadFile, HTTPException, Form, Request, Depends
+from fastapi.responses import JSONResponse
+from pydantic import BaseModel
+from typing import List, Dict, Any, Optional
+from contextlib import asynccontextmanager
+import shutil
+from fuzzywuzzy import fuzz, process
+from doctr.models import db_resnet50, crnn_vgg16_bn
+from app.doctr_module import load_custom_model, extract_names_and_boxes
+from app.pdf_ocr import extract_text_and_boxes_from_pdf
+from app.vector_database import store_text_in_vector_db
+from app.rag import retrieve_answer
+from config.config import logger
+from pathlib import Path
+import shutil
+
+
+app = FastAPI()
+
+# Define paths for the model weights
+det_weights_path = "models/db_resnet50-79bd7d70.pt"
+reco_weights_path = "models/crnn_vgg16_bn-9762b0b0.pt"
+
+# Global variables to hold the models
+det_model = None
+reco_model = None
+
+
+# Define a model for input data
+class NamePairs(BaseModel):
+    file_location: Optional[str] = None
+    names: Optional[list] = []
+
+
+# Custom dependency to handle form data and JSON
+async def get_name_pairs(
```

Comment:
No need for this to be async. The open method is not async compatible. I would rewrite the endpoint to be sync, so the request will be accessible without the await. These calls are CPU related and not much I/O related, so there is not too much improvement on creating this async endpoints. QUICK TIP: async runs all the calls in a single thread, so if you have a blocking CPU operation, the whole server will be blocked (happens a lot with inference servers). So for CPU intensive calls like performing ocr and inference on CNNs , it is preferable to set sync endpoints, where each request will spawn an independent process.